tools/sift.go

```diff
@@ -0,0 +1,556 @@
+	getSiftInvestigation,
+)
+
+// GetSiftAnalysisParams defines the parameters for retrieving a specific analysis
+type GetSiftAnalysisParams struct {
+	InvestigationID string `json:"investigationId" jsonschema:"required,description=The UUID of the investigation as a string (e.g. '02adab7c-bf5b-45f2-9459-d71a2c29e11b')"`
+	AnalysisID      string `json:"analysisId" jsonschema:"required,description=The UUID of the specific analysis to retrieve"`
+}
+
+// getSiftAnalysis retrieves a specific analysis from an investigation
+func getSiftAnalysis(ctx context.Context, args GetSiftAnalysisParams) (*analysis, error) {
+	client, err := siftClientFromContext(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("creating Sift client: %w", err)
+	}
+
+	// Parse the UUID strings
+	investigationID, err := uuid.Parse(args.InvestigationID)
+	if err != nil {
+		return nil, fmt.Errorf("invalid investigation ID format: %w", err)
+	}
+
+	analysisID, err := uuid.Parse(args.AnalysisID)
+	if err != nil {
+		return nil, fmt.Errorf("invalid analysis ID format: %w", err)
+	}
+
+	analysis, err := client.getSiftAnalysis(ctx, investigationID, analysisID)
+	if err != nil {
+		return nil, fmt.Errorf("getting analysis: %w", err)
+	}
+
+	return analysis, nil
+}
+
+// GetAnalysis is a tool for retrieving a specific analysis from an investigation
+var GetSiftAnalysis = mcpgrafana.MustTool(
+	"get_sift_analysis",
+	"Retrieves a specific analysis from an investigation by its UUID. The investigation ID and analysis ID should be provided as strings in UUID format.",
+	getSiftAnalysis,
+)
+
+// ListSiftInvestigationsParams defines the parameters for retrieving investigations
+type ListSiftInvestigationsParams struct {
+	Limit int `json:"limit,omitempty" jsonschema:"description=Maximum number of investigations to return. Defaults to 10 if not specified."`
+}
+
+// listSiftInvestigations retrieves a list of investigations with an optional limit
+func listSiftInvestigations(ctx context.Context, args ListSiftInvestigationsParams) ([]Investigation, error) {
+	client, err := siftClientFromContext(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("creating Sift client: %w", err)
+	}
+
+	// Set default limit if not provided
+	if args.Limit <= 0 {
+		args.Limit = 10
+	}
+
+	investigations, err := client.listSiftInvestigations(ctx, args.Limit)
+	if err != nil {
+		return nil, fmt.Errorf("getting investigations: %w", err)
+	}
+
+	return investigations, nil
+}
+
+// ListInvestigations is a tool for retrieving a list of investigations
+var ListSiftInvestigations = mcpgrafana.MustTool(
+	"list_sift_investigations",
+	"Retrieves a list of Sift investigations with an optional limit. If no limit is specified, defaults to 10 investigations.",
+	listSiftInvestigations,
+)
+
+// FindErrorPatternLogsParams defines the parameters for running an ErrorPatternLogs check
+type FindErrorPatternLogsParams struct {
+	Name     string            `json:"name" jsonschema:"required,description=The name of the investigation"`
```

Comment:
```suggestion
	Name     string            `json:"name" jsonschema:"required,description=The name of the investigation"`
```

I wonder if we can omit this and generate something, or if it's better to just let the model provide some random name?

Reply:
It does an ok job generating a name (see below). Maybe generating something to show the origin is better. Though we can add an investigation type for this. 🤔 
<img width="547" alt="image" src="https://github.com/user-attachments/assets/97eece7f-4364-4603-a02a-aa964316eddd" />

Reply:
Yeah it'd definitely be nice to include some info on origin/user here eventually. This is good for now though!